web/client/components/import/dragZone/enhancers/useFiles.js

```diff
@@ -3,11 +3,23 @@ const {compose, mapPropsStream, withHandlers} = require('recompose');
 
 module.exports = compose(
     withHandlers({
-        useFiles: ({ loadMap = () => { }, onClose = () => { }, setLayers = () => { } }) =>
+        useFiles: ({ currentMap, loadMap = () => { }, zoomToExtent = () => { }, onClose = () => { }, setLayers = () => { } }) =>
             ({ layers = [], maps = [] }, warnings) => {
                 const map = maps[0]; // only 1 map is allowed
                 if (map) {
-                    loadMap(map);
+                    // also handles maps without zoom or center
+                    const { zoom, center } = currentMap;
+                    loadMap({
+                        ...map,
+                        map: {
+                            ...map.map,
+                            zoom: map.map.zoom || zoom,
+                            center: map.map.center || center
+                        }
+                    });
+                    if (map.map.zoom === undefined) {
+                        zoomToExtent(map.map.maxExtent, map.map.projection, undefined, 300);
+                    }
```

Comment:
You should to externalize this into a separate epic triggered by a "loadMapFile" action, and introduce the delay there (commenting the reason) and not adding the delay to zoom to extent action.
So it's more clear that this is a side-effect limited to this context.